### PR TITLE
Cleanup conditional build logic pertaining to pre 9.2 GHCs

### PR DESCRIPTION
**Pull request by jhrcek**
Since we're supporting GHCs 9.2+ we can assume that impl(ghc >= 9.2) will be true going forward.

### CODE REVIEW

File: ghcide/ghcide.cabal [hunk 1]
```diff
@@ -232,12 +232,7 @@ library
       -Wwarn=duplicate-exports -Wwarn=dodgy-exports
       -Wwarn=incomplete-patterns -Wwarn=overlapping-patterns
       -Wwarn=incomplete-record-updates
-
-  -- ambiguous-fields is only understood by GHC >= 9.2, so we only disable it
-  -- then. The above comment goes for here too -- this should be understood to
-  -- be temporary until we can remove these warnings.
-  if (impl(ghc >=9.2) && flag(pedantic))
-    ghc-options: -Wwarn=ambiguous-fields
+      -Wwarn=ambiguous-fields
```
Comment (fendor) [inline ghcide/ghcide.cabal:235]: Shouldn't this still be guarded by the pedantic flag?

File: ghcide/ghcide.cabal [hunk 2]
```diff
@@ -269,7 +264,7 @@ executable ghcide
   hs-source-dirs:     exe
   ghc-options:
     -threaded -Wall -Wincomplete-uni-patterns -Wno-name-shadowing
-    -rtsopts "-with-rtsopts=-I0 -A128M -T"
+    -rtsopts "-with-rtsopts=-I0 -A128M -T" -Wunused-packages
```
Comment (fendor) [inline ghcide/ghcide.cabal:267]: Tiny nitpick, can we move `-Wunused-packages` to the line above, then the warnings are nicely grouped

File: plugins/hls-fourmolu-plugin/hls-fourmolu-plugin.cabal [hunk 3]
```diff
@@ -30,6 +30,7 @@ library
   build-depends:
     , base            >=4.12 && <5
     , filepath
+    , fourmolu ^>= 0.14
```
Comment (fendor) [inline plugins/hls-fourmolu-plugin/hls-fourmolu-plugin.cabal:33]:
```suggestion
    , fourmolu        ^>= 0.14
```

**Review by fendor — CHANGES_REQUESTED** [review]
LGTM, thanks for the cleanup! Just some comments or nitpicks

File: ghcide/ghcide.cabal (hunk 1 above)
Comment (jhrcek) [reply ghcide/ghcide.cabal:235]: It is - expand the code section above :)

File: ghcide/ghcide.cabal (hunk 2 above)
Comment (jhrcek) [reply ghcide/ghcide.cabal]: Applied

File: plugins/hls-fourmolu-plugin/hls-fourmolu-plugin.cabal [hunk 4]
```diff
@@ -41,13 +42,8 @@ library
     , text
     , transformers
 
-  if impl(ghc >= 9.0) && impl(ghc < 9.2)
-    build-depends: fourmolu ^>= 0.11
-  else 
-    build-depends: fourmolu ^>= 0.14
-
   -- fourmolu 0.9.0 fails to build on Windows CI for reasons unknown
```
Comment (michaelpj) [inline plugins/hls-fourmolu-plugin/hls-fourmolu-plugin.cabal:45]: the other constraint forbids fourmolu-0.9.0 already, right?

**Review by michaelpj — COMMENTED** [review]
Good stuff!

File: plugins/hls-fourmolu-plugin/hls-fourmolu-plugin.cabal (hunk 4 above)
Comment (jhrcek) [reply plugins/hls-fourmolu-plugin/hls-fourmolu-plugin.cabal:45]: You're right, I missed this one. Removed.

**Review by fendor — APPROVED** [review]
LGTM, thanks!

**Comment by jhrcek** [comment]
Rebased on master

**Pull request merged**